pkg/security/events/rate_limiter.go

```diff
@@ -102,44 +96,47 @@ func (al *AnomalyDetectionLimiter) Allow(event Event) bool {
 
 	switch category {
 	case model.ProcessCategory:
-		return al.processLimiter.Allow(event)
+		return al.processLimiter.Allow(event.GetType())
```

Comment:
With this new approach we should have only one limiter. Process and DNS event will be limiter per Workload and not anymore per event type